src/controllers/__tests__/request.test.js

```diff
@@ -0,0 +1,278 @@
+      const response = await request(server)
+        .post('/api/requests/')
+        .set('Cookie', authCookie)
+        .send(noDescriptionRequest);
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBeDefined();
+      expect(responseBody.message).toContain('Description is required');
+    });
+
+    test('Should not create a new request when image url is invalid', async () => {
+      const response = await request(server)
+        .post('/api/requests/')
+        .set('Cookie', authCookie)
+        .send(invalidPhotoRequest);
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBeDefined();
+      expect(responseBody.message).toContain('Please enter a valid image URL!');
+    });
+  });
+
+  describe('GET /api/requests/:id', () => {
+    test('Should send the correct request in the response', async () => {
+      const response = await request(server).get(`/api/requests/${RequestId}`);
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(200);
+      expect(responseBody).toMatchObject(trueRequest);
+      expect(responseBody._id.toString()).toBe(RequestId.toString());
+    });
+
+    test('Should respond with an error message when the request id is not a valid ObjectId', async () => {
+      const response = await request(server).get(
+        `/api/requests/${invalidRequestId}`
+      );
+      const responseBody = response.body;
+
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBeDefined();
+      expect(responseBody.message).toBe('Requested request ID is not valid!');
+    });
+
+    test('Should return an error message when the request does not exist', async () => {
+      const response = await request(server).get(
+        `/api/requests/${nonExistingRequestId}`
+      );
+      const responseBody = response.body;
+
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBeDefined();
+      expect(responseBody.message).toBe(
+        'The request with the specified ID was not found.'
+      );
+    });
+  });
+
+  describe('PUT /api/requests/:id', () => {
+    test('Should update name on matching request', async () => {
+      const response = await request(server)
+        .put(`/api/requests/${RequestId}`)
+        .set('Cookie', authCookie)
+        .send({ name: 'Updated name' });
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(200);
+      expect(responseBody.name).toBe('Updated name');
```

Comment:
Could you check here if the requested ID is the same with the updated request ID